api/authorization-api-1_0.md

```diff
@@ -328,77 +329,256 @@ The following is a non-normative example of a simple Decision:
+
+When used, the `obligations` context field MUST be an array of JSON objects. The obligations elements MAY follow the {{XACML}} Core specification styled URI identifiers and formats, but MAY also, and instead use custom implementation-specific strings or identifiers.
+
+If an `obligations` object is provided in the PDP response, then the PEP MUST understand it and perform all prescribed actions. 
+
+The following non-normative example depicts a single obligation to notify a manager about a denied attempt, expressed here using URI identifers:
+
+~~~ json
+[
+  {
+    "Id": "urn:example:obligation:notifyManager",
+    "AttributeAssignment": [
+      {
+        "AttributeId": "urn:example:attribute:managerEmail",
+        "Value": "manager@example.com",
+        "DataType": "string"
+      },
+      {
+        "AttributeId": "urn:example:attribute:deniedUser",
+        "Value": "bob.jones",
+        "DataType": "string"
+      },
+      {
+        "AttributeId": "urn:example:attribute:reasonForDenial",
+        "Value": "InsufficientPrivileges",
+        "DataType": "string"
+      }
+    ]
+  }
+]
+~~~
+{: #obligations-example1 title="Non-normative Example obligation with XACML Identifiers"}
+
+The following non-normative example depicts two obligations, expressed in custom implementation-specific JSON format, requiring the PEP to seek the specified `acr` and `amr` claim values for the user's `access_token` (in-effect requesting a higher level of assurance for the user's authentication; i.e., a step-up authentication), and also to email a manager about the access attempt:
+
+~~~ json
+[
+  {
+    "acr_values": "urn:com:example:loa:3",
+    "amr_values": "mfa hwk"
   },
-  "reason_user": {
-    "en-403": "Insufficient privileges. Contact your administrator",
-    "es-403": "Privilegios insuficientes. Póngase en contacto con su administrador"
+  {
+    "action": {
+      "type": "email",
+      "parameters": {
+        "to": "manager@example.com",
+        "subject": "Insufficient Assurance Level in Access Request",
+        "body": "<userid>"
+      }
+    }
   }
+]
+~~~
+{: #obligations-example2 title="Non-normative Example obligations expressed with custom values"}
+
+#### Metadata context field
+OPTIONAL - The `metadata` field is a simple JSON object, made of `key = value` pairs that can provide additional information about the request execution or the state of the PDP server or underlying services. These values can be statistics, numbers or strings, and are implementation-specific.  
+
+A PEP can ignore this field if it doesn't know how to process it.
+
+The following is a non-normative example of some Metadata context values:
+
+~~~ json
+{
+  "response-time": 60,
+  "response-time-unit": "ms",
+  "number-of-records": 2000
 }
 ~~~
-{: #example-reason-object title="Example of a Reason Object"}
+{: #metadata-example title="Non-normative Example metadata context"}
 
-### Sample Response with additional context (non-normative)
+#### Environment
```

Comment:
Again, we defined "context" in the request to provide (among other things) environmental attributes. Why are we calling these out specifically?